src/main/java/com/example/demo/service/CartService.java

```diff
@@ -0,0 +1,60 @@
+package com.example.demo.service;
+
+import com.example.demo.dto.CartDto;
+import com.example.demo.model.*;
+import com.example.demo.repository.CartRepository;
+import lombok.RequiredArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class CartService {
+    private final CartRepository cartRepository;
+    private final ModelMapper modelMapper;
+
+    public CartDto createCart(CartDto cartDto) {
+        Cart cart = modelMapper.map(cartDto, Cart.class);
+        Cart savedCart = cartRepository.save(cart);
+        return modelMapper.map(savedCart, CartDto.class);
+    }
+
+    public CartDto getCartById(Long id) {
+        Cart cart = cartRepository.findById(id).orElse(null);
+        return modelMapper.map(cart, CartDto.class);
+    }
+
+    public List<CartDto> getAllCarts() {
+        List<Cart> carts = cartRepository.findAll();
+        return carts.stream()
+                .map(cart -> modelMapper.map(cart, CartDto.class))
```

Comment:
Here you use modelmapper, but in controllers you didn't? Make sure to maintain good code and check for errors like this one.

Reply:
I have reviewed the code and made it consistent with other services. I hope that now the code is cohesive and consistent.